modules/auxiliary/gather/enum_azuresubdomains.rb

```diff
@@ -0,0 +1,155 @@
+    include Msf::Exploit::Remote::DNS::Enumeration
+
+    def initialize(info = {})
+      super(
+        update_info(
+          info,
+          'Name' => 'Azure Subdomain Scanner and Enumerator',
+          'Description' => 'This module can be used for enumerating public Azure services by locating valid subdomains through various DNS queries.',
+          'Author' => ['RoseSecurity <RoseSecurityConsulting[at]protonmail.me>'],
+          'License' => MSF_LICENSE,
+        )
+      )
+    register_options(
+        [
+          OptString.new('DOMAIN', [true, 'The target domain without TLD (Ex: victim rather than victim.org)']),
+          OptBool.new('PERMUTATIONS', [false, 'Prepend and append permutated keywords to domain (This option can take minutes to complete)', false]),
+          OptBool.new('ENUM_A', [true, 'Enumerate DNS A record', true]),
+          OptBool.new('ENUM_CNAME', [true, 'Enumerate DNS CNAME record', true]),
+          OptBool.new('ENUM_MX', [true, 'Enumerate DNS MX record', true]),
+          OptBool.new('ENUM_NS', [true, 'Enumerate DNS NS record', true]),
+          OptBool.new('ENUM_SOA', [true, 'Enumerate DNS SOA record', true]),
+          OptBool.new('ENUM_TXT', [true, 'Enumerate DNS TXT record', true]),
+        ])
+    end
+
+    def run
+      # Array of subdomains to enumerate
+      domain = datastore['DOMAIN']
+      subdomains = [ ".onmicrosoft.com",
+          ".scm.azurewebsites.net",
+          ".azurewebsites.net",
+          ".p.azurewebsites.net",
+          ".cloudapp.net",
+          ".file.core.windows.net",
+          ".blob.core.windows.net",
+          ".queue.core.windows.net",
+          ".table.core.windows.net",
+          ".mail.protection.outlook.com",
+          ".sharepoint.com",
+          ".redis.cache.windows.net",
+          ".documents.azure.com",
+          ".database.windows.net",
+          ".vault.azure.net",
+          ".azureedge.net",
+          ".search.windows.net",
+          ".azure-api.net",
+          ".azurecr.io", ]
+
+      # Array of keywords to prepend and append
+      permutations = [ "root",
+          "web",
+          "api",
+          "azure",
+          "azure-logs",
+          "data",
+          "database",
+          "data-private",
+          "data-public",
+          "dev",
+          "development",
+          "demo",
+          "files",
+          "filestorage",
+          "internal",
+          "keys",
+          "logs",
+          "private",
+          "prod",
+          "production",
+          "public",
+          "service",
+          "services",
+          "splunk",
+          "sql",
+          "staging",
+          "storage",
+          "storageaccount",
+          "test",
+          "useast",
+          "useast2",
+          "centralus",
+          "northcentralus",
+          "westcentralus",
+          "westus",
+          "westus2", ]
+
+      # Create permutated array of keywords and target domain
+      if datastore['PERMUTATIONS']
```

Comment:
There appears to be a lot of code duplication between both branches of this conditional. I suspect it could be significnatly reduced and simplified.